src/persist-client/src/cli/inspect.rs

```diff
@@ -578,8 +578,13 @@ pub async fn unreferenced_blobs(args: &StateArgs) -> Result<impl serde::Serializ
     }
 
     let mut unreferenced_blobs = UnreferencedBlobs::default();
+    let latest_version = WriterKey::for_version(&state_versions.cfg.build_version);
     for (part, writer) in all_parts {
-        if !known_writers.contains(&writer) && !known_parts.contains(&part) {
+        let is_unreferenced = match writer {
```

Comment:
This seems interesting and untested

Reply:
We don't have good automated testing for our admin CLI, but I'll check that it behaves reasonably in staging!

Reply:
This is done!